stl/inc/variant

```diff
@@ -1447,14 +1448,32 @@ _NODISCARD constexpr bool operator>=(const variant<_Types...>& _Left, const vari
     conjunction_v<is_nothrow_invocable_r<bool, greater_equal<>, const _Types&, const _Types&>...>) /* strengthened */ {
     // determine if _Left's index() is greater than _Right's, or equal and
     // _Left contains a value greater than or equal to _Right
-    using _Visitor             = _Variant_relop_visitor<greater_equal<>, _Types...>;
+    using _Visitor             = _Variant_relop_visitor2<greater_equal<>, bool, _Types...>;
     const size_t _Left_offset  = _Left.index() + 1;
     const size_t _Right_offset = _Right.index() + 1;
     return _Left_offset > _Right_offset
         || (_Left_offset == _Right_offset
             && _Variant_raw_visit(_Right_offset - 1, _Right._Storage(), _Visitor{_Left._Storage()}));
 }
 
+#ifdef __cpp_lib_concepts
+template <class... _Types>
+_NODISCARD constexpr common_comparison_category_t<compare_three_way_result_t<_Types>...>
+    operator<=>(const variant<_Types...>& _Left, const variant<_Types...>& _Right) noexcept(
+        conjunction_v<is_nothrow_invocable_r<common_comparison_category_t<compare_three_way_result_t<_Types>...>,
+            compare_three_way, const _Types&, const _Types&>...>) /* strengthened */ {

```

Comment:
Actually checking against the working draft on this pass, I see we're missing the _`requires-clause`_. Clang-format does hideous things to requires-clauses, so this will require some hand-formatting and disabling clang-format a la:
```c++
// clang-format off
template <class... _Types>
    requires (three_way_comparable<_Types> && ...)
_NODISCARD constexpr common_comparison_category_t<compare_three_way_result_t<_Types>...>
    // I'm eliding the rest of the declaration here to be lazy
            compare_three_way, const _Types&, const _Types&>...>) /* strengthened */ {
    // clang-format on
```

Reply:
Good catch, I had the requires initially but accidently removed it while testing.

Reply:
I see you have found the hidden meaning of my question 🤔